### DISCUSSION TITLE
Does the Drawer component unmount?

### DISCUSSION THREAD

**Discussion by DanishOnGit** [ant-design/ant-design#40893] [Q&A] [answered]
Hello evryone ! Plz help me in understanding Un-mounting of a component.
As far as I understand, unmounting must include removal of a component from DOM.
I am using AntD Drawer component and I have the foll code.
I want gamesAdded to be set to [ ] empty array, when I close the drawer. And hence, I wrote that in the cleanup function, so when drawer unmounts, my cleanup function setGamesAdded([ ]) will run
 
```
useEffect(() => {
    if (showDrawer) {
      setActiveTab(1);
    }
    return () => {
      setGamesAdded([]);
}
},[showDrawer])
```

But If I open and close the drawer, the drawer does not get removed from the DOM.
So, will my cleanup run??

**Comment by kiner-tang** [comment, marked as answer]
It seem that you can use `destroyOnClose ` prop to destroy drawer  when close.

**Reply by DanishOnGit** [reply]
Okkay...But that will unmount only the child components as per the docs, right?
Will the main Drawer component get removed from DOM if I include destroyOnClose?
my code is :

```
useEffect(() => {
    if (showDrawer) {
      setActiveTab(1);
    }
    return () => {
      setGamesAdded([]);
}
},[showDrawer])
```
`<Drawer visible={showDrawer} destroyOnClose/>`

**Reply by DanishOnGit** [reply]
Actually, as far as I understand, the cleanup will run @kiner-tang . But the main Drawer component can never be removed(unmounted) from DOM, right?

**Reply by kiner-tang** [reply]
Can you describe in detail what features you want to implement?

**Reply by kiner-tang** [reply]
Did you forget to update your status?
`<Drawer visible={showDrawer} destroyOnClose onClose={() => setShowDrawer(false)} />`

**Reply by DanishOnGit** [reply]
Hey @kiner-tang . Thank you so much for your response!
Actually, I am reviewing a PR and the code is working perfectly fine.
But that guy's understanding was this: cleanup is running because Drawer is getting unmounted.
So, I showed him that Drawer is not getting unmounted(since its present in the DOM, only its children are unmounting). The code works because of the `showDrawer` dependency in the deps array.
